#! /bin/sh /usr/share/dpatch/dpatch-run
## r211 | relnev | 2003-01-10 10:24:15 +0100 (Fr, 10 Jan 2003) | 2 lines
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: added keyboard updates

@DPATCH@
diff -urNad rott-1.0+dfsg~/isr.c rott-1.0+dfsg/isr.c
--- rott-1.0+dfsg~/isr.c	2008-01-09 13:52:28.000000000 +0100
+++ rott-1.0+dfsg/isr.c	2008-01-10 09:20:46.000000000 +0100
@@ -801,10 +801,9 @@
    delay=(VBLCOUNTER*delay)/10;
    IN_ClearKeysDown();
    time=GetTicCount();
-   while (GetTicCount()<time+delay)
+   while (!LastScan && GetTicCount()<time+delay)
       {
-      if (LastScan)
-         break;
+      	IN_UpdateKeyboard();
       }
 }
 
diff -urNad rott-1.0+dfsg~/rt_menu.c rott-1.0+dfsg/rt_menu.c
--- rott-1.0+dfsg~/rt_menu.c	2008-01-10 09:20:45.000000000 +0100
+++ rott-1.0+dfsg/rt_menu.c	2008-01-10 09:20:46.000000000 +0100
@@ -2948,10 +2948,10 @@
       LastScan=0;
       while (LastScan == 0)
          {
+         IN_UpdateKeyboard();
          RefreshMenuBuf( 0 );
          }
       LastScan = 0;
-      IN_UpdateKeyboard();
    }
    IN_ClearKeysDown();
    return( retval );
diff -urNad rott-1.0+dfsg~/rt_str.c rott-1.0+dfsg/rt_str.c
--- rott-1.0+dfsg~/rt_str.c	2008-01-10 09:20:44.000000000 +0100
+++ rott-1.0+dfsg/rt_str.c	2008-01-10 09:20:46.000000000 +0100
@@ -1109,6 +1109,8 @@
 //      if (GameEscaped == true)
 //         PauseLoop ();
 
+      IN_PumpEvents();
+      
       if (cursorvis)
          USL_XORICursor (x, y, xx, cursor, color);
 
